### ISSUE TITLE
mac wheels should be built for 10.9

### ISSUE THREAD

**Issue by robbuckley** [MacPython/pandas-wheels#42]
as discussed in #37, pandas wheels should really be built for macOS 10.9. The current wheels, built for 10.6, have some extensions which link to `stdlibc++`, which has been deprecated since macOS 10.9, and may no longer be installed in some case on 10.14 systems (see https://github.com/pandas-dev/pandas/issues/23424)

A quick hack solution would be to revert https://github.com/MacPython/pandas-wheels/pull/38.  This would build wheels which would break if installed on systems < 10.9. Although its highly unlikely anyone is still running such old versions of macOS.

A cleaner fix would be based on https://github.com/matthew-brett/multibuild/pull/224, which is in for review. As will as generating wheels which pip would correctly reject on pre-10.9 systems, it would reduce the gigantic filenames, from:

```
pandas-0.24.0-cp37-cp37m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
```
to

```
pandas-0.24.0-cp37-cp37m-macosx_10_9_x86_64.macosx_10_10_x86_64.whl
```
which appeals to my sense of tidyness

**Comment by jreback** [comment]
yeah we can update once your multibuild PR is merged. we can target 0.25 with >= 10.9 wheels.

**Comment by robbuckley** [comment]
quick update, i have the `multibuild` changes working, and built 0.24.0 `pandas` 10.9 wheels for python 2.7, 3.6, 3.7 using my fork. The `pandas` travis tests all pass. I am waiting for final review / merge into `multibuild`

As well as the shorter filename, and avoiding a possible recurrence of https://github.com/pandas-dev/pandas/issues/23424 for users running 10.14, as a bonus the wheels are less than half the size - around 9Mb instead of 19Mb as they are built for `x86_64` only and not dual arch.

Can you say when the first 0.25 rc is expected, and when this change needs to be ready?

**Comment by jreback** [comment]
0.25 wont be for a month or more prob 2

**Comment by robbuckley** [comment]
update, I think matthew-brett/multibuild#224 is ready now, just waiting final OK from @matthew-brett and co, then there will need to be a release of `multibuild` i suppose.  

I have built some trial `pandas 0.24` wheels with it,  they all pass the travis tests. I installed one of the new 10.9 / 64b-only wheels locally and ran the `pandas` test suite - all pass except for 1 known issue which isnt related to this. 

Build time is down from 39min for the 10.6 wheels to 28min for the 10.9, as it only has to build 64b now.

**Comment by robbuckley** [comment]
https://github.com/matthew-brett/multibuild/pull/224 has been merged to `multibuild` `devel` branch. I can raise a PR but which branch should I target ?

**Comment by jreback** [comment]
daily

**Comment by robbuckley** [comment]
before closing this ticket, is it worth making a  PR for `master` or does that wait until the next `pandas` release (0.25.0)?

**Comment by jreback** [comment]
will need to oort to master as well
can u pr there as well?

**Comment by robbuckley** [comment]
sure, will do. What about the maintenance branch?

**Comment by jreback** [comment]
actually i will just cherry pick this is fine

**Comment by robbuckley** [comment]
BTW I tested locally a couple of the daily Mac wheels after the merge of PR 46 - all good

**Comment by robbuckley** [comment]
as discussed, https://github.com/MacPython/pandas-wheels/pull/47 reinstates the changes from https://github.com/MacPython/pandas-wheels/pull/46 for daily. This version should not break the build if the new `config.sh` is used with the previous version of 'multibuild' (in that case it should fall back to the previous behaviour, building 10.6/dual arch osx wheels)
